lsp-protocol.el

```diff
@@ -225,6 +225,8 @@ Allowed params: %s" interface (reverse (-map #'cl-first params)))
                                              output-bindings)
                                        (setf current-list (cddr current-list))))))
                                  output-bindings))))
+                    `(pcase-defmacro ,interface (&rest property-bindings)
```

Comment:
Shouldn't this still add to the cases of `pcase` and cause the slowness issue?

Reply:
There's some ongoing discussion in #4430 about whether it's ok to drop the per-interface pcase right now, or whether they're part of lsp-mode's public interface and need to be deprecated for a bit first. 

If it is safe to drop these, I'll add another commit in this branch to remove this `pcase-defmacro` once I've rewritten all the `pcase`s in lsp-mode that use it.

Either way, I should add a comment to summarize all this in the code.

Reply:
Looks like the consensus is to remove the per-interface forms and let it be a breaking change. In the interest of not breaking tests, I'll fix the existing pcases and then remove this `pcase-macrodef`. :+1:

Reply:
These are gone now in 849cff06bc37bf4d7e80a293cd2ac0d83386b34b.